src/common/load-plugins.js

```diff
@@ -96,13 +96,15 @@ function loadPlugins(plugins, pluginSearchDirs) {
 function findPluginsInNodeModules(nodeModulesDir) {
   const pluginPackageJsonPaths = globby.sync(
     [
-      "prettier-plugin-*/package.json",
-      "@*/prettier-plugin-*/package.json",
-      "@prettier/plugin-*/package.json"
+      "./prettier-plugin-*/package.json",
+      "./@*/prettier-plugin-*/package.json",
+      "./@prettier/plugin-*/package.json"
     ],
-    { cwd: nodeModulesDir }
+    { cwd: nodeModulesDir, expandDirectories: false }
   );
-  return pluginPackageJsonPaths.map(path.dirname);
+  return pluginPackageJsonPaths
+    .map(packageJsonFile => path.dirname(packageJsonFile.slice(2)))
+    .sort((a, b) => a.toLowerCase().localeCompare(b.toLowerCase()));
 }
```

Comment:
something is not right here, you can work on this, restore this will pass all test,  but very slow

Reply:
Its late in China, going to sleep now, will back online in 9 hours, I  was trying to make this fast. maybe the slice is not right just need check the glob result. lets see how fast it will be

Reply:
I reverted these changes and updated `findPluginsInNodeModules` to work more like a pipeline. This should make it easier for us to add and update the post-processing steps as needed. Unfortunately I am still not understand this well enough to think of anything else useful right now.